src/triton_cli/parser.py

```diff
@@ -496,8 +465,97 @@ def parse_args(argv=None):
     parse_args_repo(subcommands)
     parse_args_server(subcommands)
     parse_args_inference(subcommands)
+    parse_args_optimize(subcommands)
     parse_args_profile(subcommands)
     parse_args_utils(subcommands)
     add_verbose_args([parser])
-    args = parser.parse_args(argv)
+
+    argv_ = argv if argv is not None else sys.argv[1:]
+    # If a passthrough command is passed as the first arg,
+    # special handling is needed.
+    need_special_handling = len(argv_) > 1 and argv_[0] in PASSTHROUGH_SUBCOMMANDS
+    if need_special_handling:
+        updated_args, pruned_args = prune_extra_subcommand_args(
+            argv_, subcommands.choices.keys()
+        )
+        args, unknown_args = parser.parse_known_args(updated_args)
+        args = add_unknown_args_to_args(args, pruned_args, unknown_args)
+    else:
+        args = parser.parse_args(argv)
     return args
+
+
+# ================================================
+# Helper functions
+# ================================================
+def build_command(executable: str, args: argparse.Namespace):
+    skip_args = ["func", "task"]
+    cmd = [executable]
+    program_subcommand = None
+    for arg, value in vars(args).items():
+        if arg in skip_args:
+            pass
+        elif arg in PASSTHROUGH_SUBCOMMANDS:
+            program_subcommand = arg
+        elif value is False:
+            pass
+        elif value is True:
+            if len(arg) == 1:
+                cmd += [f"-{arg}"]
+            else:
+                cmd += [f"--{arg}"]
+        else:
+            if len(arg) == 1:
+                cmd += [f"-{arg}", f"{value}"]
+            else:
+                arg = arg
+                cmd += [f"--{arg}", f"{value}"]
+    if program_subcommand:
+        cmd = [executable, program_subcommand] + cmd[1:]
+    return cmd
+
+
+def prune_extra_subcommand_args(argv: List[str], subcommand_names):
+    """Triton CLI can call other programs with passthrough args, so sometimes the first argument is a subcommand."""
+    """If so, prune it."""
+    pruned_args = []
+    if len(argv) > 1 and argv[1] in subcommand_names:
```

Comment:
What scenario are we guarding against here? In my testing no args are ever pruned and argv is always something like this:
`['profile', '-m', 'gpt2', '--task', 'llm', '--concurrency', '1']`

Reply:
Updated this comment to clarify. Try running `triton optimize profile <args>` (like `triton optimize profile --model-repository models \
    --profile-models add_sub --triton-launch-mode=docker \
    --output-model-repository-path output_dir \
    --export-path profile_results`). Without this logic, it will fail, since the parser thinks the profile is another subcommand.

Reply:
That's the use-case I was looking for, thanks David!

Reply:
`triton profile` and `triton optimize profile` :thinking: 

Naming things is hard..